### PR TITLE
Make number of file list entries depending on the width for grid view

### LINKED ISSUE #12295: Lazy loading of files in grid view weird
The lazy loading in grid view looks weird for the initial load of the page because my screen can show at least 32 entries and it only shows 20 items. Then on the first scroll event loads the remaining ones to fill up the space.

cc @nextcloud/designers

### PR DESCRIPTION
**Pull request by juliusknorr**
Fixes #12295 

This PR adds calculation of the number of file list entries for the available container size in grid view.

### CODE REVIEW

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -99,7 +99,14 @@
 		 * @return {int} page size
 		 */
 		pageSize: function() {
-			return Math.max(Math.ceil(this.$container.height() / 50), 1);
+			var isGridView = this.$showGridView.is(':checked');
+			var columns = 1;
+			var rows = Math.ceil(this.$container.height() / 50);
+			if (isGridView) {
+				columns = Math.ceil(this.$container.width() / 160);
```
Comment (skjnldsv) [inline apps/files/js/filelist.js]: What if later we change the size of the grid? :p 
Could we get the width by the dom elements directly? :thinking:

Comment (juliushaertl) [reply apps/files/js/filelist.js]: There are no elements in the dom at the first call, so we still would need to have a default value for that.

Comment (skjnldsv) [reply apps/files/js/filelist.js]: Make sense :)

**Review by skjnldsv — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
Code makes sense and works 👍

**Pull request merged**